### PR TITLE
SNOW-1296779: Remove sproc fallback code from Snowpark pandas

### PR DESCRIPTION
**Pull request by sfc-gh-nkrishna**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1296779, SNOW-1254730

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   This PR removes unused sproc fallback code from Snowpark pandas, not that all APIs using fallback have been replaced with NotImplementedError.

### CODE REVIEW

**Review by sfc-gh-nkrishna — COMMENTED** [review]
There are several tests in our codebase that do:
```
@pytest.mark.xfail(
    reason="SNOW-1336091: Snowpark pandas cannot run in sprocs until modin 0.28.1 is available in conda",
    strict=True,
    raises=RuntimeError,
)
@sql_count_checker(query_count=8, fallback_count=1, sproc_count=1)
```

For this PR, I have not removed these tests or updated them as they are xfailed anyhow

File: tests/integ/modin/sql_counter.py [hunk 1]
```diff
@@ -39,7 +36,6 @@
 NO_CHECK = "no_check"
 
 QUERY_COUNT_PARAMETER = "query_count"
```
Comment (sfc-gh-lspiegelberg) [inline tests/integ/modin/sql_counter.py:38]: did we remove this param in all tests?

Comment (sfc-gh-nkrishna) [reply tests/integ/modin/sql_counter.py:38]: I assume this is referring to fallback_count parameter, and yes. See this comment: https://github.com/snowflakedb/snowpark-python/pull/1542#pullrequestreview-2048434019

**Review by sfc-gh-nkrishna — COMMENTED** [review]
Kicked off this run: https://ci-dev-142.int.snowflakecomputing.com/job/SnowparkPythonSnowPandasDailyRegressRunner/401/console to confirm all tests pass

**Review by sfc-gh-helmeleegy — APPROVED** [review]
End of an era! Thanks for the final cleanup!

**Review by sfc-gh-azhan — APPROVED** [review]
LGTM

**Review by sfc-gh-aling — APPROVED** [review]

**Pull request merged**